### PR TITLE
Improve support on editing topologies workflows

### PR DESCRIPTION
**Pull request by loicalbertin**
New Features:
- Delete a workflow from topology
- Create an empty workflow in topology
- Add activities to workflows

### CODE REVIEW

File: alien4cloud/topology.go [hunk 1]
```diff
@@ -424,13 +442,55 @@ func (t *topologyService) AddNodeInA4CTopology(a4cCtx *TopologyEditorContext, No
...
+	err = t.editTopology(nil, a4cCtx, topoEditorExecute)
+
+	if err != nil {
+		return errors.Wrapf(err, "Unable to edit the topology of application '%s' and environment '%s'", a4cCtx.AppID, a4cCtx.EnvID)
+	}
+
+	return nil
+}
+
+// AddRelationship Add a new relationship in the A4C topology
```
Comment (laurentganne) [inline alien4cloud/topology.go]: wrong comment

File: alien4cloud/topology.go [hunk 2]
```diff
@@ -424,13 +442,55 @@ func (t *topologyService) AddNodeInA4CTopology(a4cCtx *TopologyEditorContext, No
...
+func (t *topologyService) CreateWorkflow(ctx context.Context, a4cCtx *TopologyEditorContext, workflowName string) error {
+	return t.createOrDeleteWorkflow(ctx, a4cCtx, "org.alien4cloud.tosca.editor.operations.workflow.CreateWorkflowOperation", workflowName)
+}
+
+// AddRelationship Add a new relationship in the A4C topology
```
Comment (laurentganne) [inline alien4cloud/topology.go]: wrong comment

File: alien4cloud/topology.go [hunk 3]
```diff
@@ -424,13 +442,55 @@ func (t *topologyService) AddNodeInA4CTopology(a4cCtx *TopologyEditorContext, No
...
+func (t *topologyService) DeleteWorkflow(ctx context.Context, a4cCtx *TopologyEditorContext, workflowName string) error {
+	return t.createOrDeleteWorkflow(ctx, a4cCtx, "org.alien4cloud.tosca.editor.operations.workflow.RemoveWorkflowOperation", workflowName)
+}
+
+// AddRelationship Add a new relationship in the A4C topology
```
Comment (laurentganne) [inline alien4cloud/topology.go]: wrong comment

**Review by laurentganne — APPROVED** [review]
Looks good!

File: alien4cloud/topology.go (hunk 1 above)
Comment (loicalbertin) [reply alien4cloud/topology.go]:
```suggestion
// CreateWorkflow creates an empty workflow in the given topology
```

File: alien4cloud/topology.go (hunk 2 above)
Comment (loicalbertin) [reply alien4cloud/topology.go]:
```suggestion
// DeleteWorkflow deletes a workflow in the given topology
```

File: alien4cloud/topology.go (hunk 3 above)
Comment (loicalbertin) [reply alien4cloud/topology.go]:
```suggestion
```

File: alien4cloud/topology.go [hunk 4]
```diff
@@ -43,6 +44,13 @@ type TopologyService interface {
 	AddRelationship(a4cCtx *TopologyEditorContext, sourceNodeName string, targetNodeName string, relType string) error
 	// Saves the topology context
 	SaveA4CTopology(a4cCtx *TopologyEditorContext) error
+	// Creates an empty workflow in the given topology
+	CreateWorkflow(ctx context.Context, a4cCtx *TopologyEditorContext, workflowName string) error
+	// Deletes a workflow in the given topology
+	DeleteWorkflow(ctx context.Context, a4cCtx *TopologyEditorContext, workflowName string) error
+
```
Comment (loicalbertin) [inline alien4cloud/topology.go]:
```suggestion
```

File: alien4cloud/topology_workflows.go [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+func (r addWorkflowActivityReq) getPreviousOperationID() string {
+	if r.PreviousOperationID == nil {
+		return ""
+	}
+	return *r.PreviousOperationID
+}
+func (r addWorkflowActivityReq) getOperationType() string {
+	return r.Type
+}
+
```
Comment (loicalbertin) [inline alien4cloud/topology_workflows.go:113]:
```suggestion

// AddWorkflowActivity adds an activity to a workflow
```

**Pull request merged**